src/VisualStudio/Core/Test/Venus/DocumentServiceTests.vb

```diff
@@ -0,0 +1,245 @@
+
+                ' make sure single line buffer doesn't throw on ExcerptMode.Tooltip
+                Dim result = Await excerptService.TryExcerptAsync(workspace.CurrentSolution.GetDocument(projectedDocument.Id), GetNamedSpan(projectedDocument), ExcerptMode.Tooltip, CancellationToken.None)
+                Assert.True(result.HasValue)
+
+                Dim content = result.Value.Content.ToString()
+
+                ' calculate expected span
+                Dim contentSpan = GetNamedSpan(subjectDocument, "Content")
+                Dim expectedContent = subjectDocument.TextBuffer.CurrentSnapshot.GetText(contentSpan.ToSpan())
+                Assert.Equal(expectedContent, content)
+
+                Dim firstText = subjectDocument.TextBuffer.CurrentSnapshot.GetText(GetNamedSpan(subjectDocument, "FirstText").ToSpan())
+                Dim lastText = subjectDocument.TextBuffer.CurrentSnapshot.GetText(GetNamedSpan(subjectDocument, "LastText").ToSpan())
+                Dim expcetedFormatted = {FormattedClassifications.Text(firstText),
+                                         Keyword("class"),
+                                         FormattedClassifications.Text(" "),
+                                         [Class]("C"),
+                                         FormattedClassifications.Text(" "),
+                                         Punctuation.OpenCurly,
+                                         FormattedClassifications.Text(" "),
+                                         Punctuation.CloseCurly,
+                                         FormattedClassifications.Text(lastText)}
+
+                Dim actualFormatted = result.Value.ClassifiedSpans.Select(Function(a) New FormattedClassification(content.Substring(a.TextSpan.Start, a.TextSpan.Length), a.ClassificationType))
+                Assert.Equal(expcetedFormatted, actualFormatted)
+
+                ' ExcerptResult.MappedSpan is relative to ExcerptResult.Content.
+                ' recalculate expected span relative to the content span
+                Dim documentSpan = GetNamedSpan(subjectDocument, "Document")
+
+                Assert.Equal(New TextSpan(documentSpan.Start - contentSpan.Start, documentSpan.Length), result.Value.MappedSpan)
+            End Using
+        End Function
+
+        Private Shared Function GetNamedSpan(document As TestHostDocument, Optional spanName As String = "Document") As TextSpan
+            Return document.AnnotatedSpans(spanName).First()
+        End Function
+
+        Private Shared Function CreateAndAddProjectionBufferDocument(workspace As TestWorkspace, subjectDocument As TestHostDocument, Optional projectedMarkup As String = Nothing) As TestHostDocument
+
+            projectedMarkup = If(projectedMarkup Is Nothing, "class projected { {|Document:|} }", projectedMarkup)
+            Dim projectedDocument = workspace.CreateProjectionBufferDocument(projectedMarkup, {subjectDocument}, LanguageNames.CSharp)
```

Comment:
I thought we had supported markup for this that @balajikris added ages ago. @CyrusNajmabadi you recall?

Reply:
possibly.  not certain myself.